src/seedu/addressbook/Main.java

```diff
@@ -105,15 +111,18 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) throws StorageOperationException, RuntimeException  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
-        } catch (Exception e) {
+        }   catch (StorageOperationException e) {
+                throw e;
```

Comment:
Is it required to explicitly catch and through this here without doing any additional handling?